pdp/ast/yast/selector.go

```diff
@@ -7,59 +7,65 @@ import (
 	"github.com/infobloxopen/themis/pdp"
 )
 
-func (ctx context) unmarshalSelector(v interface{}) (pdp.LocalSelector, boundError) {
+func (ctx context) unmarshalSelector(v interface{}) (pdp.Expression, boundError) {
+	var ret pdp.Expression
+
 	m, err := ctx.validateMap(v, "selector attributes")
 	if err != nil {
-		return pdp.LocalSelector{}, err
+		return ret, err
 	}
 
-	s, err := ctx.extractString(m, yastTagURI, "selector URI")
+	uri, err := ctx.extractString(m, yastTagURI, "selector URI")
 	if err != nil {
-		return pdp.LocalSelector{}, err
+		return ret, err
 	}
 
-	ID, ierr := url.Parse(s)
+	id, ierr := url.Parse(uri)
 	if ierr != nil {
-		return pdp.LocalSelector{}, newSelectorURIError(s, ierr)
+		return ret, newSelectorURIError(uri, ierr)
 	}
 
-	if strings.ToLower(ID.Scheme) == "local" {
-		loc := strings.Split(ID.Opaque, "/")
-		if len(loc) != 2 {
-			return pdp.LocalSelector{}, newSelectorLocationError(ID.Opaque, s)
-		}
+	items, err := ctx.extractList(m, yastTagPath, "path")
+	if err != nil {
+		return ret, bindErrorf(err, "selector(%s)", id.Opaque)
+	}
 
-		items, err := ctx.extractList(m, yastTagPath, "path")
+	path := make([]pdp.Expression, len(items))
+	for i, item := range items {
+		e, err := ctx.unmarshalExpression(item)
 		if err != nil {
-			return pdp.LocalSelector{}, bindErrorf(err, "selector(%s.%s)", loc[0], loc[1])
+			return ret, bindErrorf(bindErrorf(err, "%d", i), "selector(%s)", id.Opaque)
 		}
 
-		path := make([]pdp.Expression, len(items))
-		for i, item := range items {
-			e, err := ctx.unmarshalExpression(item)
-			if err != nil {
-				return pdp.LocalSelector{}, bindErrorf(bindErrorf(err, "%d", i), "selector(%s.%s)", loc[0], loc[1])
-			}
+		path[i] = e
+	}
 
-			path[i] = e
-		}
+	st, err := ctx.extractString(m, yastTagType, "type")
+	if err != nil {
+		return ret, bindErrorf(err, "selector(%s)", id.Opaque)
+	}
 
-		s, err := ctx.extractString(m, yastTagType, "type")
-		if err != nil {
-			return pdp.LocalSelector{}, bindErrorf(err, "selector(%s.%s)", loc[0], loc[1])
-		}
+	t, ok := pdp.TypeIDs[strings.ToLower(st)]
```

Comment:
would be nice to move common jast/yast logic into separate function to avoid copying of code